kyo-core/shared/src/test/scala/kyoTest/internal/TraceTest.scala

```diff
@@ -0,0 +1,284 @@
+package kyoTest.internal
+
+import kyo.internal.Trace
+import kyoTest.KyoTest
+
+class TraceTest extends KyoTest:
+
+    "show" in {
+        def test(i: Int)(using t: Trace) = t.show
+        assert(test(42) ==
+            """TraceTest.scala:10
+            |test(42)""".stripMargin)
+    }
+
+    "no param" in {
+        def test(using t: Trace) = t.snippet
+        assert(test == "test")
+    }
+
+    "one pram" in {
+        def test(i: Int)(using t: Trace) = t.snippet
+        assert(test(42) == "test(42)")
+    }
+
+    "two params" in {
+        def test(a: Int, b: String)(using t: Trace) = t.snippet
+        assert(test(1, "hello") == "test(1, \"hello\")")
+    }
+
+    "multiple param lists" in {
+        def test(a: Int)(b: String)(using t: Trace) = t.snippet
+        assert(test(1)("hello") == "test(1)(\"hello\")")
+    }
+
+    "infix method" in pendingUntilFixed {
+        object Test:
+            infix def test(v: Int)(using t: Trace) = t.snippet
+
+        assert((Test test 42) == "Test test 42")
```

Comment:
How would this be serialized differently?

Maybe an extension infix method might be a good test as well.

Reply:
> How would this be serialized differently?

Note that the test is pending. The macro is currently generating an empty string for this scenario.

> Maybe an extension infix method might be a good test as well.

I'll add a pending test for it as well.